packages/async-flow/src/types.d.ts

```diff
@@ -78,15 +81,23 @@ export type GuestInterface<T> = {
  *
  * Specifically, Promise return values are converted to Vows.
  */
-export type HostOf<F extends CallableFunction> = F extends (
+export type HostFn<F extends CallableFunction> = F extends (
   ...args: infer A
 ) => infer R
   ? R extends Promise<infer T>
-    ? (...args: A) => Vow<T extends Passable ? T : HostInterface<T>>
-    : (...args: A) => HostInterface<R>
-  : F;
-
-export type HostArgs<GA extends any[]> = { [K in keyof GA]: HostOf<GA[K]> };
+    ? (...args: HostArgs<A>) => Vow<T extends Passable ? T : HostInterface<T>>
+    : (...args: HostArgs<A>) => HostInterface<R>
+  : never;
+
+export type HostOf<T extends Passable> = T extends CallableFunction
+  ? HostFn<T>
+  : T extends CopyRecord
+    ? T
+    : HostInterface<T>;
```

Comment:
Wondering if we shouldn't blindly map to `HostInterface` regardless of matching `CopyRecord`. The reason is to be able to map nested values within the copy record which can contain Remote and Promise